go/vm/lfvm/instructions_test.go

```diff
@@ -285,3 +285,59 @@ func TestCreateChecksBalance(t *testing.T) {
 	}
 
 }
+
+func TestCreateShanghaiInitCodeSize(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	runContext := vm.NewMockRunContext(ctrl)
+
+	source := vm.Address{1}
+	ctxt := context{
+		status: RUNNING,
+		params: vm.Parameters{
+			Recipient: source,
+		},
+		context:  runContext,
+		stack:    NewStack(),
+		memory:   NewMemory(),
+		gas:      50000,
+		revision: vm.R12_Shanghai,
+	}
+
+	// Prepare stack arguments.
+	ctxt.stack.stack_ptr = 3
+	ctxt.stack.data[0].Set(uint256.NewInt(49153)) // < size of the init code
```

Comment:
maybe add a comment explaining where does this number comes from ? or use `MaxInitCodeSize+1`